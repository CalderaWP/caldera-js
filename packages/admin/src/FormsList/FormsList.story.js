import React from 'react';
import { storiesOf } from '@storybook/react';
import { FormsList } from './FormsList';
//import {forms} from '../../EntryViewer/forms.fixtures';

const forms = [
	{
		id: 'contact-form',
		name: 'Contact Form'
	},
	{
		id: 'other-form',
		name: 'Other Form'
	}
];
storiesOf('FormsList', module).add('No forms', () => (
	<FormsList
		forms={[]}
		panelTitle={'Panel Title'}
		noFormsMessage={'Custom No Forms Found'}
		onFormAction={() => {}}
	/>
));

storiesOf('FormsList', module).add('With forms', () => (
	<FormsList
		forms={Object.values(forms)}
		panelTitle={'Panel Title'}
		noFormsMessage={'Custom No Forms Found'}
		onFormAction={() => {}}
	/>
));
