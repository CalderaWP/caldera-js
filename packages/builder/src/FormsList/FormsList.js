import React,{useState,useMemo,useEffect,Fragment} from 'react';
import {Panel, PanelBody, PanelRow} from '@wordpress/components';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {FormListItem} from '../FormListItem/FormListItem';
import {fieldAreaFactory,fieldFactory} from "@calderajs/components";
import moment from 'moment';
const fuzzysearch = function(needle, haystack) {
    if( ! haystack.length || ! needle.length ){
        return [];
    }
    const results = [];
    haystack.forEach(item  => {
        if( needle === item || item.includes(needle) ){
            results.push(item);
        }
    });
    return  results;
};

/**
 *
 * @param forms
 * @param panelTitle
 * @param classname
 * @param onFormAction
 * @return {*}
 * @constructor
 */
export const FormsList = ({forms, panelTitle, classname, onFormAction}) => {
    const [searchBy,setSearchBy] = useState('');
    const [sortedForms,setSortedForms] = useState(forms);
    const [sortBy,setSortBy] = useState('name');
    const [sortOrder,setSortOrder] = useState('DESC');



    const getFormKey = (form,key,defaultValue) => form.hasOwnProperty(key) ? form[key] : defaultValue;

    useEffect( () => {
       doSort()
    },[sortBy,forms,sortOrder,searchBy]);


    const getFormNames = () => {
        let names = [];
        forms.forEach(form => {names.push(form.name)});
        return names;
    };

    const doSort = () => {
        let sorted = forms;
        let _forms = forms;
        if( searchBy && 2 < searchBy.length ){
            const foundFormNames = fuzzysearch( searchBy, getFormNames() );
            if( ! foundFormNames.length ){
                setSortedForms([]);
                return;
            }else{
                _forms = forms.filter( form => foundFormNames.includes(form.name) );
            }
        }

        switch (sortBy) {
            case '_last_updated':
                sorted = _forms.sort((form,lastForm) => {
                    const d1 = getFormKey(form,sortBy,null).replace(/[^\+]*$/, '').replace( ' +', '' );
                    const d2 = getFormKey(lastForm,sortBy,null).replace(/[^\+]*$/, '').replace( ' +', '' );
                    const currentMoment = moment(d1);
                    const lastMoment = moment(d2);
                    if ('ASC' === sortOrder) {
                        if( currentMoment.isBefore(lastMoment) ){
                            return -1;
                        }else{
                            return 1;
                        }
                    }
                    if( ! currentMoment.isBefore(lastMoment) ){
                        return -1;
                    }else{
                        return 1;
                    }
                    return  ! currentMoment.isBefore(lastMoment);
                });
                break;
            case 'name':
            default:
                sorted = _forms.sort((form,lastForm) => {
                    const current = getFormKey(form, 'name', null);
                    const last = getFormKey(lastForm, 'name', null);
                    if ('ASC' === sortOrder) {
                        return last.localeCompare(current);
                    }
                    return current.localeCompare(last);


                });
        break;
        }
        setSortedForms(sorted);
    };

    const sortField = {
        fieldType: 'dropdown',
        label: 'Sort Forms By',
        value: sortBy,
        options: [
            {
                value: 'name',
                label : 'Name'
            },
            {
                value: '_last_updated',
                label : 'Updated'
            },
        ]
    };
    const orderField = {
        fieldType: 'dropdown',
        label: 'Order Forms By',
        value: sortOrder,
        options: [
            {
                value: 'ASC',
                label : 'name' === sortBy ? 'Alphabetical' : 'Most Recently Updated',
            },
            {
                value: 'DESC',
                label : 'name' === sortBy ? 'Reverse Alphabetical' : 'Least Recently Updated First',
            },
        ]
    };

    const searchField = {
        fieldType: 'text',
        label: 'Search Forms By',
        value: searchBy,

    };



    const FormSearch = () => {
        return (
            <Fragment>
                {fieldAreaFactory(searchField, setSearchBy )}
                {fieldAreaFactory(sortField, setSortBy)}
                {fieldAreaFactory(orderField, setSortOrder)}

            </Fragment>
        )
    };


    return (
        <Panel classname={classNames(classname)}>
            <PanelBody title={panelTitle} icon="feedback" initialOpen={true}>
                <FormSearch />
                {sortedForms.map(form => {
                    return (
                        <PanelRow key={form.id}>
                            <FormListItem
                                key={form.id}
                                form={form}
                                onFormAction={onFormAction}
                            />
                        </PanelRow>
                    );
                })}
            </PanelBody>
        </Panel>
    )
};

/**
 * Prop type definitions for the FormsList component
 *
 * @type {{forms: shim, panelTitle: shim, noFormsMessage: shim, onFormAction: *}}
 */
FormsList.propTypes = {
    forms: PropTypes.array,
    panelTitle: PropTypes.string,
    noFormsMessage: PropTypes.string,
    onFormAction: PropTypes.func.isRequired
};

/**
 * Default props for the forms list component.
 *
 * @type {{forms: Array, panelTitle: string, noFormsMessage: string}}
 */
FormsList.defaultProps = {
    forms: [],
    panelTitle: 'All Forms',
    noFormsMessage: 'No Forms Found'
};
