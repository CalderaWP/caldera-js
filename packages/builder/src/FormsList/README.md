# Forms List Component
Displays a list of forms

## Usage
`import {FormsList} from '@calderajs/components';`

### Listing Forms
```jsx
const forms = [
    {
        id: 'form-id',
        name: 'Contact Form'
    },
    {
        //...
    }
];
<FormsList
    forms={forms}
    panelTitle={'Mailchimp Forms'}
    noFormsMessage={'No Forms Found'}
    onFormAction={(formId,actionName) => {
        console.log(`Action name ${actioName} called for form ${formId}` )
    }}
/>

```

### items prop
Optionally, an array of icons can be passed to replace the default set.

```
const items = [
	{
		icon: 'edit',
		label: 'Edit Form',
		actionName: 'edit',
	},
	{
		icon: 'list-view',
		label: 'View Entries',
		actionName: 'view-entries',
	},
];
<FormsList
    items={items}
    forms={forms}
    panelTitle={'Mailchimp Forms'}
    noFormsMessage={'No Forms Found'}
    onFormAction={(formId,actionName) => {
        console.log(`Action name ${actioName} called for form ${formId}` )
    }}
/>
```

### `onFormAction()`
See [`FormListItem`](../FormListItem/README.md) component.
