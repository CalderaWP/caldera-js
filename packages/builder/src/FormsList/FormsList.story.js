import React from "react";
import { storiesOf } from "@storybook/react";
import { FormsList } from "./FormsList";

const forms = [
	{
		id: "contact-form",
		name: "Contact Form",
	},
	{
		id: "other-form",
		name: "Other Form",
	},
];
storiesOf("FormsList", module).add("No forms", () => (
	<FormsList
		forms={[]}
		panelTitle={"Panel Title"}
		noFormsMessage={"Custom No Forms Found"}
		onFormAction={() => {}}
	/>
));

storiesOf("FormsList", module).add("With forms", () => (
	<FormsList
		forms={Object.values(forms)}
		panelTitle={"Panel Title"}
		noFormsMessage={"Custom No Forms Found"}
		onFormAction={(formId, actionName) => {
			console.log(formId, actionName);
		}}
	/>
));
