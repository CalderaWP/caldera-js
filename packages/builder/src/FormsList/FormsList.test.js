import * as React from "react";
import { render } from "react-testing-library";
import { FormsList } from "./FormsList";

describe("FormsList", () => {
	const forms = [
		{
			id: "contact-form",
			name: "Contact Form",
		},
		{
			id: "other-form",
			name: "Other Form",
		},
	];
	let onFormAction;
	beforeEach(() => {
		onFormAction = jest.fn();
	});

	it("Matches snapshot ", () => {
		expect(
			render(
				<FormsList
					forms={forms}
					panelTitle={"Panel Title"}
					noFormsMessage={"Custom No Forms Found"}
					onFormAction={() => {}}
				/>
			)
		).toMatchSnapshot();
	});
});
