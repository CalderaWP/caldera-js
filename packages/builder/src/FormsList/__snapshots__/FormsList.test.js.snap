// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormsList Matches snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="components-panel"
      >
        <div
          class="components-panel__body is-opened"
        >
          <h2
            class="components-panel__body-title"
          >
            <button
              aria-expanded="true"
              class="components-button components-panel__body-toggle"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                <svg
                  aria-hidden="true"
                  class="components-panel__arrow"
                  focusable="false"
                  height="24px"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M0,0h24v24H0V0z"
                      fill="none"
                    />
                  </g>
                  <g>
                    <path
                      d="M12,8l-6,6l1.41,1.41L12,10.83l4.59,4.58L18,14L12,8z"
                    />
                  </g>
                </svg>
              </span>
              Panel Title
              <svg
                aria-hidden="true"
                class="dashicon dashicons-feedback components-panel__icon"
                focusable="false"
                height="20"
                role="img"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 2h16c.55 0 1 .45 1 1v14c0 .55-.45 1-1 1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1zm15 14V7H3v9h14zM4 8v1h3V8H4zm4 0v3h8V8H8zm-4 4v1h3v-1H4zm4 0v3h8v-3H8z"
                />
              </svg>
            </button>
          </h2>
          <div
            class="caldera-field-group caldera-field-area-button form-group"
          >
            <label
              class="form-label"
            >
               
            </label>
            <button
              class="btn btn-secondary"
              fieldtype="button"
              type="button"
              value="Reset Search"
              wrapperclassnames="caldera-field-group,caldera-field-area-button"
            >
              Reset Search
            </button>
          </div>
          <div
            class="caldera-field-group caldera-field-area-text form-group"
          >
            <label
              class="form-label"
            >
              Search Forms By Name
               
            </label>
            <input
              class="caldera-field caldera-field-text form-control"
              type="text"
              value=""
            />
          </div>
          <div
            class="caldera-field-group caldera-field-area-dropdown form-group"
          >
            <label
              class="form-label"
            >
              Sort Forms By
               
            </label>
            <select
              class="caldera-field caldera-field-select form-control"
            >
              <option
                value="name"
              >
                Name
              </option>
              <option
                value="_last_updated"
              >
                Updated
              </option>
            </select>
            
          </div>
          <div
            class="caldera-field-group caldera-field-area-dropdown form-group"
          >
            <label
              class="form-label"
            >
              Order Forms By
               
            </label>
            <select
              class="caldera-field caldera-field-select form-control"
            >
              <option
                value="ASC"
              >
                Alphabetical
              </option>
              <option
                value="DESC"
              >
                Reverse Alphabetical
              </option>
            </select>
            
          </div>
          <div
            class="components-panel__row"
          >
            <div>
              <div>
                Contact Form
              </div>
              <div
                class="components-toolbar"
              >
                <button
                  aria-label="Edit Form"
                  class="components-button components-icon-button form-list-item form-list-item-edit"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-edit"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.89 3.39l2.71 2.72c.46.46.42 1.24.03 1.64l-8.01 8.02-5.56 1.16 1.16-5.58s7.6-7.63 7.99-8.03c.39-.39 1.22-.39 1.68.07zm-2.73 2.79l-5.59 5.61 1.11 1.11 5.54-5.65zm-2.97 8.23l5.58-5.6-1.07-1.08-5.59 5.6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="View Entries"
                  class="components-button components-icon-button form-list-item form-list-item-view-entries"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-list-view"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2 19h16c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1v16c0 .55.45 1 1 1zM4 3c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V3h11zM4 7c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V7h11zM4 11c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11zM4 15c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Preview Form"
                  class="components-button components-icon-button form-list-item form-list-item-preview"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-feedback"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2 2h16c.55 0 1 .45 1 1v14c0 .55-.45 1-1 1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1zm15 14V7H3v9h14zM4 8v1h3V8H4zm4 0v3h8V8H8zm-4 4v1h3v-1H4zm4 0v3h8v-3H8z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Form Settings"
                  class="components-button components-icon-button form-list-item form-list-item-settings"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-admin-settings"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M18 16V4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h13c.55 0 1-.45 1-1zM8 11h1c.55 0 1 .45 1 1s-.45 1-1 1H8v1.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V13H6c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V11zm5-2h-1c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V7h1c.55 0 1 .45 1 1s-.45 1-1 1h-1v5.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V9z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="components-panel__row"
          >
            <div>
              <div>
                Other Form
              </div>
              <div
                class="components-toolbar"
              >
                <button
                  aria-label="Edit Form"
                  class="components-button components-icon-button form-list-item form-list-item-edit"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-edit"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.89 3.39l2.71 2.72c.46.46.42 1.24.03 1.64l-8.01 8.02-5.56 1.16 1.16-5.58s7.6-7.63 7.99-8.03c.39-.39 1.22-.39 1.68.07zm-2.73 2.79l-5.59 5.61 1.11 1.11 5.54-5.65zm-2.97 8.23l5.58-5.6-1.07-1.08-5.59 5.6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="View Entries"
                  class="components-button components-icon-button form-list-item form-list-item-view-entries"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-list-view"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2 19h16c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1v16c0 .55.45 1 1 1zM4 3c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V3h11zM4 7c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V7h11zM4 11c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11zM4 15c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Preview Form"
                  class="components-button components-icon-button form-list-item form-list-item-preview"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-feedback"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2 2h16c.55 0 1 .45 1 1v14c0 .55-.45 1-1 1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1zm15 14V7H3v9h14zM4 8v1h3V8H4zm4 0v3h8V8H8zm-4 4v1h3v-1H4zm4 0v3h8v-3H8z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Form Settings"
                  class="components-button components-icon-button form-list-item form-list-item-settings"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-admin-settings"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M18 16V4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h13c.55 0 1-.45 1-1zM8 11h1c.55 0 1 .45 1 1s-.45 1-1 1H8v1.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V13H6c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V11zm5-2h-1c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V7h1c.55 0 1 .45 1 1s-.45 1-1 1h-1v5.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V9z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="components-panel"
    >
      <div
        class="components-panel__body is-opened"
      >
        <h2
          class="components-panel__body-title"
        >
          <button
            aria-expanded="true"
            class="components-button components-panel__body-toggle"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              <svg
                aria-hidden="true"
                class="components-panel__arrow"
                focusable="false"
                height="24px"
                role="img"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0,0h24v24H0V0z"
                    fill="none"
                  />
                </g>
                <g>
                  <path
                    d="M12,8l-6,6l1.41,1.41L12,10.83l4.59,4.58L18,14L12,8z"
                  />
                </g>
              </svg>
            </span>
            Panel Title
            <svg
              aria-hidden="true"
              class="dashicon dashicons-feedback components-panel__icon"
              focusable="false"
              height="20"
              role="img"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 2h16c.55 0 1 .45 1 1v14c0 .55-.45 1-1 1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1zm15 14V7H3v9h14zM4 8v1h3V8H4zm4 0v3h8V8H8zm-4 4v1h3v-1H4zm4 0v3h8v-3H8z"
              />
            </svg>
          </button>
        </h2>
        <div
          class="caldera-field-group caldera-field-area-button form-group"
        >
          <label
            class="form-label"
          >
             
          </label>
          <button
            class="btn btn-secondary"
            fieldtype="button"
            type="button"
            value="Reset Search"
            wrapperclassnames="caldera-field-group,caldera-field-area-button"
          >
            Reset Search
          </button>
        </div>
        <div
          class="caldera-field-group caldera-field-area-text form-group"
        >
          <label
            class="form-label"
          >
            Search Forms By Name
             
          </label>
          <input
            class="caldera-field caldera-field-text form-control"
            type="text"
            value=""
          />
        </div>
        <div
          class="caldera-field-group caldera-field-area-dropdown form-group"
        >
          <label
            class="form-label"
          >
            Sort Forms By
             
          </label>
          <select
            class="caldera-field caldera-field-select form-control"
          >
            <option
              value="name"
            >
              Name
            </option>
            <option
              value="_last_updated"
            >
              Updated
            </option>
          </select>
          
        </div>
        <div
          class="caldera-field-group caldera-field-area-dropdown form-group"
        >
          <label
            class="form-label"
          >
            Order Forms By
             
          </label>
          <select
            class="caldera-field caldera-field-select form-control"
          >
            <option
              value="ASC"
            >
              Alphabetical
            </option>
            <option
              value="DESC"
            >
              Reverse Alphabetical
            </option>
          </select>
          
        </div>
        <div
          class="components-panel__row"
        >
          <div>
            <div>
              Contact Form
            </div>
            <div
              class="components-toolbar"
            >
              <button
                aria-label="Edit Form"
                class="components-button components-icon-button form-list-item form-list-item-edit"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-edit"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.89 3.39l2.71 2.72c.46.46.42 1.24.03 1.64l-8.01 8.02-5.56 1.16 1.16-5.58s7.6-7.63 7.99-8.03c.39-.39 1.22-.39 1.68.07zm-2.73 2.79l-5.59 5.61 1.11 1.11 5.54-5.65zm-2.97 8.23l5.58-5.6-1.07-1.08-5.59 5.6z"
                  />
                </svg>
              </button>
              <button
                aria-label="View Entries"
                class="components-button components-icon-button form-list-item form-list-item-view-entries"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-list-view"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 19h16c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1v16c0 .55.45 1 1 1zM4 3c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V3h11zM4 7c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V7h11zM4 11c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11zM4 15c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11z"
                  />
                </svg>
              </button>
              <button
                aria-label="Preview Form"
                class="components-button components-icon-button form-list-item form-list-item-preview"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-feedback"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 2h16c.55 0 1 .45 1 1v14c0 .55-.45 1-1 1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1zm15 14V7H3v9h14zM4 8v1h3V8H4zm4 0v3h8V8H8zm-4 4v1h3v-1H4zm4 0v3h8v-3H8z"
                  />
                </svg>
              </button>
              <button
                aria-label="Form Settings"
                class="components-button components-icon-button form-list-item form-list-item-settings"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-admin-settings"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 16V4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h13c.55 0 1-.45 1-1zM8 11h1c.55 0 1 .45 1 1s-.45 1-1 1H8v1.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V13H6c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V11zm5-2h-1c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V7h1c.55 0 1 .45 1 1s-.45 1-1 1h-1v5.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V9z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="components-panel__row"
        >
          <div>
            <div>
              Other Form
            </div>
            <div
              class="components-toolbar"
            >
              <button
                aria-label="Edit Form"
                class="components-button components-icon-button form-list-item form-list-item-edit"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-edit"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.89 3.39l2.71 2.72c.46.46.42 1.24.03 1.64l-8.01 8.02-5.56 1.16 1.16-5.58s7.6-7.63 7.99-8.03c.39-.39 1.22-.39 1.68.07zm-2.73 2.79l-5.59 5.61 1.11 1.11 5.54-5.65zm-2.97 8.23l5.58-5.6-1.07-1.08-5.59 5.6z"
                  />
                </svg>
              </button>
              <button
                aria-label="View Entries"
                class="components-button components-icon-button form-list-item form-list-item-view-entries"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-list-view"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 19h16c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1v16c0 .55.45 1 1 1zM4 3c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V3h11zM4 7c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V7h11zM4 11c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11zM4 15c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11z"
                  />
                </svg>
              </button>
              <button
                aria-label="Preview Form"
                class="components-button components-icon-button form-list-item form-list-item-preview"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-feedback"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 2h16c.55 0 1 .45 1 1v14c0 .55-.45 1-1 1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1zm15 14V7H3v9h14zM4 8v1h3V8H4zm4 0v3h8V8H8zm-4 4v1h3v-1H4zm4 0v3h8v-3H8z"
                  />
                </svg>
              </button>
              <button
                aria-label="Form Settings"
                class="components-button components-icon-button form-list-item form-list-item-settings"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-admin-settings"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 16V4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h13c.55 0 1-.45 1-1zM8 11h1c.55 0 1 .45 1 1s-.45 1-1 1H8v1.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V13H6c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V11zm5-2h-1c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V7h1c.55 0 1 .45 1 1s-.45 1-1 1h-1v5.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V9z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
